src/components/checkbox/checkbox.test.js

```diff
@@ -0,0 +1,557 @@
+
+      checkboxRole()
+        .focus()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    it("should call onClick callback when a click event is triggered", () => {
+      CypressMountWithProviders(<CheckboxComponent onClick={callback} />);
+
+      checkboxRole()
+        .click()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    describe("Testing CheckboxGroup component", () => {
+      it.each(specialCharacters)(
+        "should render CheckboxGroup component with %s as legend",
+        (legendValue) => {
+          CypressMountWithProviders(
+            <CheckboxGroupComponent legend={legendValue} />
+          );
+          legend().should("have.text", legendValue);
+        }
+      );
+
+      it("should render CheckboxGroup component with error ", () => {
+        CypressMountWithProviders(<CheckboxGroupComponent error />);
+
+        cy.get("svg").should(
+          "have.css",
+          "border-bottom-color",
+          "rgb(205, 56, 75)"
+        );
+      });
+
+      it("should render CheckboxGroup component with warning ", () => {
+        CypressMountWithProviders(<CheckboxGroupComponent warning />);
+
+        cy.get("svg").should(
+          "have.css",
+          "border-bottom-color",
+          "rgb(239, 103, 0)"
+        );
+      });
+
+      it("should render CheckboxGroup component with info ", () => {
+        CypressMountWithProviders(<CheckboxGroupComponent info />);
+
+        cy.get("svg").should(
+          "have.css",
+          "border-bottom-color",
+          "rgb(0, 96, 167)"
+        );
+      });
+
+      it("should render CheckboxGroup component with error message ", () => {
+        CypressMountWithProviders(
+          <CheckboxGroupComponent error="Error has occurred" />
+        );
+
+        errorIcon()
+          .should("exist")
+          .should("have.attr", "aria-label")
+          .and("contain", "Error has occurred");
+      });
+
+      it("should render CheckboxGroup component with warning message", () => {
+        CypressMountWithProviders(
+          <CheckboxGroupComponent warning="Warning has occurred" />
+        );
+
+        warningIcon()
+          .should("exist")
+          .should("have.attr", "aria-label")
+          .and("contain", "Warning has occurred");
+      });
+
+      it("should render CheckboxGroup component with info message", () => {
+        CypressMountWithProviders(
+          <CheckboxGroupComponent info="Info has occurred" />
+        );
+
+        infoIcon()
+          .should("exist")
+          .should("have.attr", "aria-label")
+          .and("contain", "Info has occurred");
+      });
+
+      it("should render CheckboxGroup component with groupName", () => {
+        CypressMountWithProviders(<CheckboxGroupComponent />);
```

Comment:
where you have set it?

Reply:
I set groupname when mounting the component